readers/hdfs/src/main/java/com/criteo/hadoop/garmadon/hdfs/offset/FsBasedCheckpointer.java

```diff
@@ -0,0 +1,68 @@
+package com.criteo.hadoop.garmadon.hdfs.offset;
+
+import org.apache.hadoop.fs.FSDataOutputStream;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+/**
+ * Write checkpoints on the filesystem.
+ */
+public class FsBasedCheckpointer implements Checkpointer {
+    private static final Logger LOGGER = LoggerFactory.getLogger(FsBasedCheckpointer.class.getName());
+
+    private final FileSystem fs;
+    private final Map<Path, Boolean> checkpointsCache;
```

Comment:
The cache is never cleaned. Even if it is not containing a lot of data I don't like the idea that eventually we fill memory with useless data until the JVM dies.

Reply:
We're talking about nb_events x uptime days, it would take a very long time to be of a significative size...while I agree with you on the principle, I would qualify this as premature optimization

Reply:
When you will take in account partitions, that will be more data. 
Cleaning the cache is not about optimisation, it is more about managing memory properly and avoid obvious leaks. Is there such thing as "premature correct memory management" ?

Reply:
@gray-matter you can use guva cache for this (already loaded by hadoop dependencies)
I don't se this as a premature optimization but as a bug
Design which rely on probable restart of instances before the OOM issue doesn't looks good to me

Reply:
Done